From: Josh Poimboeuf <jpoimboe@kernel.org>
Date: Tue, 8 Apr 2025 00:02:15 -0700
Subject: objtool: Stop UNRET validation on UD2
Git-commit: 9f9cc012c2cbac4833746a0182e06a8eec940d19
Patch-mainline: v6.15-rc2
References: git-fixes

In preparation for simplifying INSN_SYSCALL, make validate_unret()
terminate control flow on UD2 just like validate_branch() already does.

Signed-off-by: Josh Poimboeuf <jpoimboe@kernel.org>
Signed-off-by: Ingo Molnar <mingo@kernel.org>
Cc: Linus Torvalds <torvalds@linux-foundation.org>
Link: https://lore.kernel.org/r/ce841269e7e28c8b7f32064464a9821034d724ff.1744095216.git.jpoimboe@kernel.org

Acked-by: Miroslav Benes <mbenes@suse.cz>
---
 tools/objtool/check.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/tools/objtool/check.c
+++ b/tools/objtool/check.c
@@ -4006,6 +4006,9 @@ static int validate_unret(struct objtool
 			break;
 		}
 
+		if (insn->dead_end)
+			return 0;
+
 		if (!next) {
 			WARN_INSN(insn, "teh end!");
 			return -1;
